app/src/main/java/com/hedvig/app/feature/offer/ui/OfferActivity.kt

```diff
@@ -363,7 +359,9 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
                     model.onOpenCheckout()
                 }
             }
-            CheckoutMethod.APPROVE_ONLY -> model.approveOffer()
+            CheckoutMethod.APPROVE_ONLY -> lifecycleScope.launch {
+                model.approveOffer()
+            }
```

Comment:
Are we sure we want to bind the job handling this to the view lifecycle coroutine scope instead of the viewmodel coroutine scope?
I mean like having the function not be suspending and instead launching a coroutine from the viewmodel instead like we usually do because it lives longer than the lifecycle one.

Reply:
Fixing!